src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  const copyOfState = JSON.parse(JSON.stringify(state));
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const item in action.extraData) {
+          copyOfState[item] = action.extraData[item];
+        }
+        resultArray.push({ ...copyOfState });
+        break;
```

Comment:
You repeat this part. [Checklist](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) p.4

Reply:
i push element on every step. if each case doesn't work, my switch will break without any actions. if i'll delete push actions from every case and put him to the floor of loop - it will break my function